### ISSUE TITLE
“overrides... but not marked 'override'” warnings in build on macOS

### ISSUE THREAD

**Issue by apjanke** [lxqt/qterminal#564]
##### Expected Behavior

Warning-free build.

##### Current Behavior

When building on macOS against Qt 5 with Xcode 10.1 (clang 10.0), I'm seeing several warnings like the following in my `make` output.

```
/Users/janke/local/repos/qterminal/src/tabwidget.h:107:10: warning: 'contextMenuEvent' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
    void contextMenuEvent(QContextMenuEvent * event);
         ^
/usr/local/opt/qt/lib/QtWidgets.framework/Headers/qwidget.h:630:18: note: overridden virtual function is here
    virtual void contextMenuEvent(QContextMenuEvent *event);
                 ^
/Users/janke/local/repos/qterminal/src/tabwidget.h:114:10: warning: 'eventFilter' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
    bool eventFilter(QObject *obj, QEvent *event);
         ^
/usr/local/opt/qt/lib/QtCore.framework/Headers/qobject.h:128:18: note: overridden virtual function is here
    virtual bool eventFilter(QObject *watched, QEvent *event);
                 ^
2 warnings generated.
/Users/janke/local/repos/qterminal/src/termwidget.h:95:14: warning: 'paintEvent' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
        void paintEvent (QPaintEvent * event);
             ^
/usr/local/opt/qt/lib/QtWidgets.framework/Headers/qwidget.h:625:18: note: overridden virtual function is here
    virtual void paintEvent(QPaintEvent *event);
                 ^
1 warning generated.
```
##### Possible Solution

This looks like a new C++11 thing.

Maybe you need to add explicit `override` modifiers to the member functions? Like:

```
bool eventFilter(QObject *obj, QEvent *event);
```

Or if you're not going to use them, turn off the `inconstent-missing-override` warning with `-Wno-inconsistent-missing-override`.

##### System Information

* Distribution & Version: macOS 10.13.6, Xcode 10.1
* Kernel: Darwin Kernel Version 17.7.0
* Qt Version: 5.12.1
* libqtxdg Version: N/A 
* qtermwidget Version: `master`
* lxqt-build-tools Version: 0.6.0
* Package version: ???

**Comment by yan12125** [comment]
> Maybe you need to add explicit override modifiers to the member functions?

Correct!

**Comment by apjanke** [comment]
Another question: these methods are not declared `virtual` in the derived class. Shouldn't they be `virtual`? Or are you using some special static-dispatch technique here?

**Comment by yan12125** [comment]
I bet Qt devs are doing well and already declare functions to be overridden as `virtual` in base classes :)

**Comment by apjanke** [comment]
> I bet Qt devs are doing well and already declare functions to be overridden as virtual in base classes :)

You're right. I forgot that C++ doesn't require you to redeclare them as `virtual` in the derived class.

**Comment by apjanke** [comment]
PR to fix: #567
